test/unit/FieldsetDate.test.js

```diff
@@ -0,0 +1,151 @@
+
+      expect(container.find('label').at(0).text()).to.equal('Start Year');
+      expect(container.find('label').at(0).find('input').length).to.equal(1);
+      expect(container.find('label').at(0).find('input').props().type).to.equal('text');
+      expect(container.find('label').at(0).find('input').props().maxLength).to.equal('4');
+
+      expect(container.find('label').at(1).text()).to.equal('End Year');
+      expect(container.find('label').at(1).find('input').length).to.equal(1);
+      expect(container.find('label').at(1).find('input').props().type).to.equal('text');
+      expect(container.find('label').at(1).find('input').props().maxLength).to.equal('4');
+    });
+
+    it('should have the default state of dateAfter and dateBefore of "0".', () => {
+      expect(component.state('dateAfter')).to.equal('0');
+      expect(component.state('dateBefore')).to.equal('0');
+    });
+
+    it('should have the default state of selectedFacets as the props passed.', () => {
+      expect(component.state('selectedFacets')).to.equal(selectedFacets);
+    });
+
+    it('should render no error messages', () => {
+      expect(component.find('#error-message').text()).to.equal('');
+    });
+  });
+
+  describe('It updates the store', () => {
+    let component;
+    let updateSelectedFacets;
+    const selectedFacets = {
+      language: ['lang:fr', 'lang:fr'],
+    };
+    const calledWithStartDateValues = {
+      language: ['lang:fr', 'lang:fr'],
+      dateAfter: '2001',
+    };
+    const calledWithEndDateValues = {
+      language: ['lang:fr', 'lang:fr'],
+      dateAfter: '2001',
+      dateBefore: '2100',
+    };
+
+    before(() => {
+      component = mount(<FieldsetDate selectedFacets={selectedFacets} />);
+      updateSelectedFacets = sinon.spy(Actions, 'updateSelectedFacets');
+    });
+
+    afterEach(() => {
+      updateSelectedFacets.reset();
+    });
+
+    after(() => {
+      updateSelectedFacets.restore();
+      component.unmount();
+    });
+
+    it('should not update the state if no valid target name exists.', () => {
+      const startYearInput = component.find('#input-container').find('label').at(0).find('input');
+
+      startYearInput.simulate('change', { target: { name: 'test-name', value: '2001' } });
+
+      expect(component.state('dateAfter')).to.equal('0');
+      expect(component.state('dateBefore')).to.equal('0');
+      expect(updateSelectedFacets.calledOnce).to.equal(false);
+    });
+
+    it('should update selectedFacets based on it\'s input from Start Year input.', () => {
+      const startYearInput = component.find('#input-container').find('label').at(0).find('input');
+
+      startYearInput.simulate('change', { target: { name: 'start-date', value: '2001' } });
+
+      expect(component.state('dateAfter')).to.equal('2001');
+      expect(updateSelectedFacets.calledOnce).to.equal(true);
+      expect(component.state('selectedFacets')).to.deep.equal(calledWithStartDateValues);
```

Comment:
You can also do `.to.eql`. It's the same as `.to.deep.equal`.

Reply:
`.to.eql` and `.to.equal` are different but means `.to.deep.equal`?

Reply:
Yep. It's a shorthand for `to.deep.equal`.

Reply:
Good to know! How do you think about the short hand? I think I still prefer `to.deep.equal`. The syntax of the short hand is too much like `.to.equal`'s typo.